pilot/pkg/model/push_context.go

```diff
@@ -1839,16 +1839,59 @@ func (ps *PushContext) initSidecarScopes(env *Environment) {
 	// Root namespace can have only one sidecar config object
 	// Currently we expect that it has no workloadSelectors
 	var rootNSConfig *config.Config
-	ps.sidecarIndex.sidecarsByNamespace = make(map[string][]*SidecarScope, len(sidecarConfigs))
 	for i, sidecarConfig := range sidecarConfigs {
-		ps.sidecarIndex.sidecarsByNamespace[sidecarConfig.Namespace] = append(ps.sidecarIndex.sidecarsByNamespace[sidecarConfig.Namespace],
-			ConvertToSidecarScope(ps, &sidecarConfig, sidecarConfig.Namespace))
-		if rootNSConfig == nil && sidecarConfig.Namespace == ps.Mesh.RootNamespace &&
+		if sidecarConfig.Namespace == ps.Mesh.RootNamespace &&
```

Comment:
nit: would prefer to keep the old code exactly when ConvertSidecarScopeConcurrency==1 so we don't need to drop down into the possibly tricky concurrency code at all

Reply:
Thanks, done.